node/rpc-client/src/lib.rs

```diff
@@ -1,12 +1,180 @@
+/// Filecoin RPC client interface methods
+pub mod auth_ops;
+pub mod chain_ops;
+pub mod net_ops;
+pub mod wallet_ops;
+
+/// Filecoin HTTP JSON-RPC client methods
+use jsonrpc_v2::{Error, Id, RequestObject, V2};
+use log::{debug, error};
+use parity_multiaddr::{Multiaddr, Protocol};
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+use std::env;
+
+const DEFAULT_MULTIADDRESS: &str = "/ip4/127.0.0.1/tcp/1234/http";
+const DEFAULT_URL: &str = "http://127.0.0.1:1234/rpc/v0";
+const DEFAULT_PROTOCOL: &str = "http";
+const DEFAULT_HOST: &str = "127.0.0.1";
+const DEFAULT_PORT: &str = "1234";
+const API_INFO_KEY: &str = "FULLNODE_API_INFO";
+const RPC_ENDPOINT: &str = "rpc/v0";
+
+/// Error object in a response
+#[derive(Deserialize)]
+pub struct JsonRpcError {
+    pub code: i64,
+    pub message: String,
+}
+
+#[derive(Deserialize)]
+#[serde(untagged)]
+pub enum JsonRpcResponse<R> {
+    Result {
+        jsonrpc: V2,
+        result: R,
+        id: Id,
+    },
+    Error {
+        jsonrpc: V2,
+        error: JsonRpcError,
+        id: Id,
+    },
+}
+
+struct URL {
+    protocol: String,
+    port: String,
+    host: String,
+}
+
+/// Parses a multiaddress into a URL
+fn multiaddress_to_url(ma_str: String) -> String {
+    // Parse Multiaddress string
+    let ma: Multiaddr = ma_str.parse().expect("Parse multiaddress");
+
+    // Fold Multiaddress into a URL struct
+    let addr = ma.into_iter().fold(
+        URL {
+            protocol: DEFAULT_PROTOCOL.to_owned(),
+            port: DEFAULT_PORT.to_owned(),
+            host: DEFAULT_HOST.to_owned(),
+        },
+        |mut addr, protocol| {
+            match protocol {
+                Protocol::Ip6(ip) => {
+                    addr.host = ip.to_string();
+                }
+                Protocol::Ip4(ip) => {
+                    addr.host = ip.to_string();
+                }
+                Protocol::Dns(dns) => {
+                    addr.host = dns.to_string();
+                }
+                Protocol::Dns4(dns) => {
+                    addr.host = dns.to_string();
+                }
+                Protocol::Dns6(dns) => {
+                    addr.host = dns.to_string();
+                }
+                Protocol::Dnsaddr(dns) => {
+                    addr.host = dns.to_string();
+                }
+                Protocol::Tcp(port) => {
+                    addr.port = port.to_string();
+                }
+                Protocol::Http => {
+                    addr.protocol = "http".to_string();
+                }
+                Protocol::Https => {
+                    addr.protocol = "https".to_string();
+                }
+                _ => {}
+            };
+            addr
+        },
+    );
+
+    // Format, print and return the URL
+    let url = format!(
+        "{}://{}:{}/{}",
+        addr.protocol, addr.host, addr.port, RPC_ENDPOINT
+    );
+    debug!("Using JSON-RPC v2 HTTP URL: {}", url);
```

Comment:
Should debug log this where this method actually gets called.

Reply:
How would you suggest I do that?